src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs

```diff
@@ -726,6 +726,35 @@ RequestContext<object> requestContext
                     this.editorSession,
                     eventContext);
             }
+
+            // Convert the editor file glob patterns into an array for the Workspace
+            // TODO Is this thread-safe? this is hardly an atomic operation. But intermediate objects
+            // are costly. Given it's a config only change maybe a race condition won't occur?
+            editorSession.Workspace.ExcludeFilesGlob.Clear();
+            if (configChangeParams.Settings.Files != null && configChangeParams.Settings.Files.Exclude != null)
+            {
+                foreach(string pattern in configChangeParams.Settings.Files.Exclude.Keys)
```

Comment:
What do these key value pairs look like? I find it odd that you're doing a `foreach` for the keys and then using `configChangeParams.Settings.Files.Exclude[pattern]`.

Reply:
The hash looks like this:  (thus the check)

```
  "files.exclude": {
    "**/.git": true,
    "**/.svn": true,
    "**/.hg": true,
    "**/CVS": true,
    "**/.DS_Store": true
  },
```

Reply:
So I guess they wanted it to be in a Set for the quick lookup?

So bizarre having the `: true` for each one instead of just a list.

Reply:
🤷‍♂️ Watchya gonna do? That's what we have.